internal/service/quicksight/dashboard.go

```diff
@@ -203,7 +203,7 @@ func resourceDashboardRead(ctx context.Context, d *schema.ResourceData, meta int
 		return diag.FromErr(err)
 	}
 
-	out, err := FindDashboardByID(ctx, conn, d.Id())
+	out, err := FindDashboardByID(ctx, conn, d.Id(), -1)
```

Comment:
Do we always want the latest version as this is hardcoded?

Reply:
That is the current (as in before this PR) behaviour, so no change to the current behaviour.

That said, we could easily add a version parameter to this `resourceDashboardRead()` function if that would be beneficial for the future?

Reply:
The issue this MR tries to fix is urgent since the current provider version makes it impossible to update a dashboard. 

To update a dashboard we currently use the workaround to rename the dashboard, hence deleting the old and creating a new dashboard. This has the impact of making the user data, such as bookmarks, customisation, etc, vanish for the user and therefore not making it very user-friendly.

@dev-slatto I struggle to understand the concern you are raising. The old version of the code also tries to update to the latest version and it does not even allow you to update the dashboard at all since it now throws the error referred to in this MR.

When deploying a new version it would be (in my opinion) natural to deploy the latest version in most of the use cases. I struggle to see a good use case for not enabling the latest version except for maybe testing the new version before it is published to everyone. We have separate dev/test/prod env, so for us, it makes sense to always publish the latest.

Reply:
> Do we always want the latest version as this is hardcoded?

That is the current (as in before this PR) behaviour, so no change to the current behaviour.

That said, we could easily add a version parameter to this `resourceDashboardRead()` function if that would be beneficial for the future?

Reply:
Had a chat with Graham and we concluded that it makes sence for a hardcoded version for the resource, and that a data source of dashboard should be able to include the version paramater. However, see the comment later down in the PR about defining the version as a constant instead of hardcoding it 😄

Reply:
Great, I have updated to use a constant 👍